internal/livelint/check_can_visit_public_app_internal_test.go

```diff
@@ -0,0 +1,178 @@
+package livelint
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/matryer/is"
+	apiv1 "k8s.io/api/core/v1"
+	netv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	v1 "k8s.io/client-go/kubernetes/typed/networking/v1"
+)
+
+var services = []apiv1.Service{
+	{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "service1",
+			Namespace: "namespace",
+		},
+		Spec: apiv1.ServiceSpec{
+			Selector: map[string]string{},
+			Ports: []apiv1.ServicePort{
+				{
+					Port:       80,
+					Protocol:   apiv1.ProtocolTCP,
+					TargetPort: intstr.FromInt(8080),
+					Name:       "http",
+				},
+			},
+		},
+	},
+}
+
+var ingresses = []netv1.Ingress{
+	{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ingress",
+			Namespace: "namespace",
+		},
+		Spec: netv1.IngressSpec{
+			Rules: []netv1.IngressRule{
+				{
+					Host: "host.test.com",
+					IngressRuleValue: netv1.IngressRuleValue{
+						HTTP: &netv1.HTTPIngressRuleValue{
+							Paths: []netv1.HTTPIngressPath{
+								{
+									Path:     "/",
+									PathType: &pathTypePrefix,
+									Backend: netv1.IngressBackend{
+										Service: &netv1.IngressServiceBackend{
+											Name: "service1",
+											Port: netv1.ServiceBackendPort{
+												Number: 80,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+}
+
+var pathTypePrefix = netv1.PathTypePrefix
+
+var errConnection = errors.New("unable to resolve host")
+
+type RoundTripFunc func(req *http.Request) (*http.Response, error)
+
+func (f RoundTripFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req)
+}
+
+func NewTestClient(fn RoundTripFunc) *http.Client {
```

Comment:
Should we put these types and methods into a file like `http_test.go`? I think this file here is a bit overloaded, and it's quite likely that they will be reused in other tests.

Reply:
Done. Had to call the file `http_internal_test.go` though.